app/controllers/users_controller.rb

```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+# display infornation about, and edit, users
+class UsersController < ApplicationController
+  # These need to be moved into the files where they are actually used.
+  require "find"
+  require "set"
+
+  before_action :login_required
+  before_action :disable_link_prefetching
+
+  # User index, restricted to admins.
+  def index
+    return patterned_index if params[:pattern].present?
+
+    if in_admin_mode? || find_query(:User)
+      query = find_or_create_query(:User, by: params[:by])
+      show_selected_users(query, id: params[:id].to_s, always_index: true)
+    else
+      flash_error(:runtime_search_has_expired.t)
+      redirect_to("/")
+    end
+  end
+
+  alias index_user index
+  # People guess this page name frequently for whatever reason, and
+  # since there is a view with this name, it crashes each time.
+  alias list_users index
+
+  # User index, restricted to admins.
+  def by_name
```

Comment:
Would this be better as a flavor or type of `index`, getting rid of one path? 
(Sorry for the belated comment. Maybe the above should be a separate PR, so as not to delay this PR.)

Reply:
Yes, it is possible that GET /users?by=name is equivalent.  Likewise, below, GET /users?by=contribution could replace /users/contribution or /users/contributors.

Reply:
Car's in the shop! It was a failure of the cooling system, recently worked on.

My arbitrary philosophy with "flavors of index" so far has been, if it's different arrangements of data using the same layout, make the flavor a param of the index path. But if it's a completely different layout, using different partials for example, use a different path. 

Going by that, i feel like contributors, which seems like a different presentation of user data from the index, should be a separate path. Actually, the most hassle-free way to refactor it would be to make it a separate "Contributors" controller with one route: index. This would be my preferred 'path'.  i.e., mo.org/contributors

What do you guys think?

Reply:
Another way of saying that tentative "flavor-of-index-philosophy" is 

"if you can (or would) switch between these views of the data on the same view", like filtering the index.